server/swimos_introspection/src/task/mod.rs

```diff
@@ -111,6 +116,46 @@ pub fn init_introspection(
     (resolver, meta_agent, task)
 }
 
+/// Trait for routing tables with which the introspection agents can be registered.
+pub trait AgentRegistration {
+    /// Register an agent type.
+    ///
+    /// # Arguments
+    /// * `pattern` - The route pattern at which the agent will be loaded.
+    /// * `agent` - The agent type.
```

Comment:
```suggestion
    /// * `agent` - The agent to register.
```
As it's accepting an instance.